### DISCUSSION TITLE
Reolink Doorbell issues

### DISCUSSION THREAD

**Discussion by tibmeister** [koush/scrypted#1211] [Q&A] [answered]
I have two wired Reolink cameras and just purchased a Reolink WiFi Doorbell.  The wired cameras have worked pretty good, but the doorbell seems to not want to work with the plugin.  Here's a copy of the logs, and everything is up to date, firmware, plugins, etc.

One thing that seems to be prevalent is _listen loop connection failed, restarting listener. Wrong ONVIF SOAP response_, but below is the logs from a "default" config.

[Rebroadcast Plugin]: RTMP main.bcs prebuffer session started
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session starting
stream url rtmp://192.168.1.25:1935/bcs/channel0_main.bcs?channel=0&stream=0&user=admin&password=XXredactedXX
[Rebroadcast Plugin]: -hide_banner -fflags +genpts -i rtmp:[REDACTED] -rtsp_transport tcp -vcodec copy -acodec copy -f rtsp rtsp://127.0.0.1:43853/2eccfee35b4d2df1
[Rebroadcast Plugin]: rtmp://admin:XXredactedXX@192.168.1.25:1935/bcs/channel0_main.bcs?channel=0&stream=0&user=admin&password=XXredactedXX: Input/output error
[Rebroadcast Plugin]: ffmpeg exited
[Rebroadcast Plugin]: rebroadcast error Error: ffmpeg exited
[Rebroadcast Plugin]:     at ChildProcess.<anonymous> (/@scrypted/prebuffer-mixin/main.nodejs.js:1:76444)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:events:526:35)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:domain:489:12)
[Rebroadcast Plugin]:     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session ended with error Error: ffmpeg killed before sdp could be parsed
[Rebroadcast Plugin]:     at /@scrypted/prebuffer-mixin/main.nodejs.js:1:76761
[Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Rebroadcast Plugin]: Reolink Doorbell restarting prebuffer session in 5 seconds
listen loop connection failed, restarting listener. Wrong ONVIF SOAP response
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session started
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session starting
stream url rtmp://192.168.1.25:1935/bcs/channel0_main.bcs?channel=0&stream=0&user=admin&password=XXredactedXX
[Rebroadcast Plugin]: -hide_banner -fflags +genpts -i rtmp:[REDACTED] -rtsp_transport tcp -vcodec copy -acodec copy -f rtsp rtsp://127.0.0.1:43079/d1561ad8d19a75d8
[Rebroadcast Plugin]: rtmp://admin:XXredactedXX@192.168.1.25:1935/bcs/channel0_main.bcs?channel=0&stream=0&user=admin&password=XXredactedXX: Input/output error
[Rebroadcast Plugin]: ffmpeg exited
[Rebroadcast Plugin]: rebroadcast error Error: ffmpeg exited
[Rebroadcast Plugin]:     at ChildProcess.<anonymous> (/@scrypted/prebuffer-mixin/main.nodejs.js:1:76444)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:events:526:35)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:domain:489:12)
[Rebroadcast Plugin]:     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session ended with error Error: ffmpeg killed before sdp could be parsed
[Rebroadcast Plugin]:     at /@scrypted/prebuffer-mixin/main.nodejs.js:1:76761
[Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Rebroadcast Plugin]: Reolink Doorbell restarting prebuffer session in 5 seconds
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session started
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session starting
stream url rtmp://192.168.1.25:1935/bcs/channel0_main.bcs?channel=0&stream=0&user=admin&password=XXredactedXX
[Rebroadcast Plugin]: -hide_banner -fflags +genpts -i rtmp:[REDACTED] -rtsp_transport tcp -vcodec copy -acodec copy -f rtsp rtsp://127.0.0.1:46539/27f0f4773f63cd71
[Rebroadcast Plugin]: rtmp://admin:XXredactedXX@192.168.1.25:1935/bcs/channel0_main.bcs?channel=0&stream=0&user=admin&password=XXredactedXX: Input/output error
[Rebroadcast Plugin]: ffmpeg exited
[Rebroadcast Plugin]: rebroadcast error Error: ffmpeg exited
[Rebroadcast Plugin]:     at ChildProcess.<anonymous> (/@scrypted/prebuffer-mixin/main.nodejs.js:1:76444)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:events:526:35)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:domain:489:12)
[Rebroadcast Plugin]:     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session ended with error Error: ffmpeg killed before sdp could be parsed
[Rebroadcast Plugin]:     at /@scrypted/prebuffer-mixin/main.nodejs.js:1:76761
[Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Rebroadcast Plugin]: Reolink Doorbell restarting prebuffer session in 5 seconds
listen loop connection failed, restarting listener. Wrong ONVIF SOAP response
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session started
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session starting
stream url rtmp://192.168.1.25:1935/bcs/channel0_main.bcs?channel=0&stream=0&user=admin&password=XXredactedXX
[Rebroadcast Plugin]: -hide_banner -fflags +genpts -i rtmp:[REDACTED] -rtsp_transport tcp -vcodec copy -acodec copy -f rtsp rtsp://127.0.0.1:34153/3a6f69fd29fd0252
[Rebroadcast Plugin]: rtmp://admin:XXredactedXX@192.168.1.25:1935/bcs/channel0_main.bcs?channel=0&stream=0&user=admin&password=XXredactedXX: Input/output error
[Rebroadcast Plugin]: ffmpeg exited
[Rebroadcast Plugin]: rebroadcast error Error: ffmpeg exited
[Rebroadcast Plugin]:     at ChildProcess.<anonymous> (/@scrypted/prebuffer-mixin/main.nodejs.js:1:76444)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:events:526:35)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:domain:489:12)
[Rebroadcast Plugin]:     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session ended with error Error: ffmpeg killed before sdp could be parsed
[Rebroadcast Plugin]:     at /@scrypted/prebuffer-mixin/main.nodejs.js:1:76761
[Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Rebroadcast Plugin]: Reolink Doorbell restarting prebuffer session in 5 seconds
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session started
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session starting
stream url rtmp://192.168.1.25:1935/bcs/channel0_main.bcs?channel=0&stream=0&user=admin&password=XXredactedXX
[Rebroadcast Plugin]: -hide_banner -fflags +genpts -i rtmp:[REDACTED] -rtsp_transport tcp -vcodec copy -acodec copy -f rtsp rtsp://127.0.0.1:36947/2e1788a3646850d3
[Rebroadcast Plugin]: rtmp://admin:XXredactedXX@192.168.1.25:1935/bcs/channel0_main.bcs?channel=0&stream=0&user=admin&password=XXredactedXX: Input/output error
[Rebroadcast Plugin]: ffmpeg exited
[Rebroadcast Plugin]: rebroadcast error Error: ffmpeg exited
[Rebroadcast Plugin]:     at ChildProcess.<anonymous> (/@scrypted/prebuffer-mixin/main.nodejs.js:1:76444)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:events:526:35)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:domain:489:12)
[Rebroadcast Plugin]:     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session ended with error Error: ffmpeg killed before sdp could be parsed
[Rebroadcast Plugin]:     at /@scrypted/prebuffer-mixin/main.nodejs.js:1:76761
[Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Rebroadcast Plugin]: Reolink Doorbell restarting prebuffer session in 5 seconds
listen loop connection failed, restarting listener. Wrong ONVIF SOAP response

**Comment by koush** [comment, marked as answer]
use a simple password

**Reply by tibmeister** [reply]
So we can't have a good secure password on these devices?  Is it certain special chars we need to avoid?

**Reply by EagleStClair** [reply]
> So we can't have a good secure password on these devices?  Is it certain special chars we need to avoid?

Did that even solve the issue? I'm having the same error message when using a simple password. Reolink camera with dummy doorbell button to act as doorbell.

**Reply by tibmeister** [reply]
It did work without any special chars.

**Comment by koush** [comment]
numbers and letters are fine, but it seems to have issues with special characters when embedded into an RTMP url.

**Reply by tibmeister** [reply]
Only numbers and letters now, reloaded the plugin and seeing the same results.

**Reply by tibmeister** [reply]
I also went ahead and created a limited user in the camera with a pretty simple password so that my admin account can still have a very strong password.  Even with the simple password, 14 chars, upper and lowercase and numbers, same console entries and no functionality.

**Reply by tibmeister** [reply]
One last thing before I call it a night.  I opened the URL from the console log in VLC and it pulls the stream without issue, so confident the setup isn't an issue at this point, just not sure what the plugin is looking for beyond the stream.

**Reply by tibmeister** [reply]
It seems to work despite some of the messages that make it seem like there's an error going on.  Doorbell events don't work, and I've noticed motion events on this and my other Reolink cameras no longer work as well.
